Cannot Install RethinkDB using Binaries on Ubuntu 16.10

After doing the following command:

```bash
source /etc/lsb-release && echo "deb http://download.rethinkdb.com/apt $DISTRIB_CODENAME main" | sudo tee /etc/apt/sources.list.d/rethinkdb.list
wget -qO- https://download.rethinkdb.com/apt/pubkey.gpg | sudo apt-key add -
```

I make an update and I got error:
```bash
Err:8 http://download.rethinkdb.com/apt yakkety/main amd64 Packages            
  404  Not Found
...
...
...
Fetched 93.3 kB in 12s (7,632 B/s)                                             
Reading package lists... Done
W: The repository 'http://download.rethinkdb.com/apt yakkety Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch http://download.rethinkdb.com/apt/dists/yakkety/main/binary-amd64/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.
casper@casper-Z97X-UD3H:~$ sudo apt-get -y install rethinkdb
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package rethinkdb
```

How can I solve this problem?

I tried to install RethinkDB from source. After I execute the command 
```
make -j 4
```
and I got:

```bash
    [1/421] CC build/release/obj/rdb_protocol/ql2.pb.o
    [2/421] INSTALL-INCLUDE re2_20140111
    [3/421] INSTALL-INCLUDE v8_3.30.33.16
    [4/421] INSTALL-INCLUDE jemalloc_4.1.0
    [5/421] CP build/release/rethinkdb-gdb.py
    [6/421] CONVERT_PROTOFILE build/drivers/ruby/lib/ql2.pb.rb
    [7/421] CONVERT_PROTOFILE drivers/ruby/lib/ql2.pb.rb
    [9/421] CONVERT_PROTOFILE build/drivers/python/rethinkdb/ql2_pb2.py
    [8/421] CP ./build/drivers/python/rethinkdb/
    [10/421] CONVERT_PROTOFILE drivers/python/rethinkdb/ql2_pb2.py
    [11/421] BUILD jemalloc_4.1.0
    [12/421] BUILD re2_20140111
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* VersionDummy::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:613:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Query_AssocPair::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:907:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Query::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:1324:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Frame::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:2011:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Backtrace::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:2404:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Response::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:2946:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Datum_AssocPair::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:3592:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Datum::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:4018:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Term_AssocPair::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:5053:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
build/proto/rdb_protocol/ql2.pb.cc: In member function ‘virtual google::protobuf::uint8* Term::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const’:
build/proto/rdb_protocol/ql2.pb.cc:5406:10: error: unused parameter ‘deterministic’ [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
src/build.mk:385: recipe for target 'build/release/obj/rdb_protocol/ql2.pb.o' failed
make[1]: *** [build/release/obj/rdb_protocol/ql2.pb.o] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:52: recipe for target 'make' failed
make: *** [make] Error 2
```

--- a/mk/packaging.mk
+++ b/mk/packaging.mk
@@ -42,8 +42,16 @@ DSC_CONFIGURE_DEFAULT = --prefix=/usr --sysconfdir=/etc --localstatedir=/var
 DIST_CONFIGURE_DEFAULT_FETCH = $(foreach pkg, $(DIST_SUPPORT_PACKAGES), --fetch $(pkg))
 DIST_SUPPORT = $(foreach pkg, $(DIST_SUPPORT_PACKAGES), $(SUPPORT_SRC_DIR)/$(pkg)_$($(pkg)_VERSION))
 
-DEB_BUILD_DEPENDS := g++, libboost-dev, libssl-dev, curl, m4, debhelper
+DEB_BUILD_DEPENDS := libboost-dev, libssl-dev, curl, m4, debhelper
 DEB_BUILD_DEPENDS += , fakeroot, python, libncurses5-dev, libcurl4-openssl-dev, libssl-dev
+
+ifeq ($(UBUNTU_RELEASE),yakkety)
+  # RethinkDB fails to compile with GCC 6 (#5757)
+  DEB_BUILD_DEPENDS += , g++-5
+  DSC_CONFIGURE_DEFAULT += CXX=g++-5
+else
+  DEB_BUILD_DEPENDS += , g++
+endif
 
 ifneq (1,$(BUILD_PORTABLE))
   DEB_BUILD_DEPENDS += , protobuf-compiler, libprotobuf-dev

